File: C++/ex2.cpp
#include <iostream>
#include <string>

void modification(std::string* var){
    // (*var).append(" le monde");
    var->append(" le monde");
    return;
}


int main()
{
    std::string str = "Bonjour";
    modification(&str);
    std::cout << str << std::endl;
    return 0;
}

File: TypeScript/ex2.ts
//--------------------------------------------------------------

// les references en typescript ça n'existe pas

//--------------------------------------------------------------

function modification(variable:string): string{
    return variable + " le monde";
}

function ex2() : void
{
    let str = "Bonjour";
    str = modification(str);
    console.log(str);
    return;
}

ex2();